blockchain/pool.go

```diff
@@ -453,7 +458,7 @@ type bpRequester struct {
 	pool       *BlockPool
 	height     int64
 	gotBlockCh chan struct{}
-	redoCh     chan struct{}
+	redoCh     chan p2p.ID //redo may send multitime, add peerId to identify repeat
```

Comment:
As we only remember last requester, I don't see how this helps. Maybe it would be useful to remember what are peers we already asked block for in case bad peer reconnect, but I am not sure that this suggestion address that case.